client/reader/id-helpers.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * External Dependencies
+ */
+
+/**
+ * Internal Dependencies
+ */
+
+export function toValidId( val ) {
+	if ( val === true ) {
+		return undefined;
+	}
```

Comment:
is it okay to turn `false` into `0`?

what did you think of using the positive case to validate, not "if it's any of these values reject" but instead "only if it's any of these values allow"

```js
const type = typeof val;

if ( 
	! val || // not defined
	( 'string' !== type && 'number' !== type ) || // cannot otherwise represent numbers
	( 'string' === type && /\./.test( val ) ) // contains a fractional part
) {
	return undefined;
}

const num = Number( val );

return ( 'number' === typeof num && isFinite( num ) )
	? num
	: undefined;
```